### PR TITLE
new state for applications that are delay waiting

### PR DESCRIPTION
**Pull request by gkodali-zededa**
Signed-off-by: Gopi krishna Kodali <gkodali@zededa.com>

### CODE REVIEW

File: api/proto/info/info.proto [hunk 1]
```diff
@@ -201,6 +201,12 @@ enum ZSwState {
   // state. If an App Instance is in ERROR state, it means there is NO DOMAIN
   // currently running.
   ERROR                 = 21;   // Error State
+
+  // It would have been ideal to leave a gap before the ERROR state.
+  // Since we cannot change the existing states now, introduce the
+  // Z_SW_STATE_START_DELAYED state after ERROR.
+  // Had to prefix with Z_SW_STATE_ to satisfy yetus checks.
```
Comment (eriknordmark) [inline api/proto/info/info.proto:208]: We tend to ignore the yetus complaints when adding an enum value to an existing enum to instead have them be internally consistent, so START_DELAYED makes more sense.

As we implement this does it make sense to use the Progress to indicate the percentage of time which is left?

File: api/proto/info/info.proto [hunk 2]
```diff
@@ -201,6 +201,12 @@ enum ZSwState {
   // state. If an App Instance is in ERROR state, it means there is NO DOMAIN
   // currently running.
   ERROR                 = 21;   // Error State
+
+  // It would have been ideal to leave a gap before the ERROR state.
+  // Since we cannot change the existing states now, introduce the
+  // Z_SW_STATE_START_DELAYED state after ERROR.
```
Comment (eriknordmark) [inline api/proto/info/info.proto:207]: We have code which tracks min state in zedmanager and/or volumemgr (mostly about the volumes that go through downloading, verifying, etc) so we can report the minimum in the app instance status. 
I don't think there are other places and states where their numeric values matter. If they do we can have the internal state be in numeric order and have the function which maps to ZSwState take care of it.

**Review by eriknordmark — CHANGES_REQUESTED** [review]
Remove the Z_SW_STATE prefix since this is a new value in an existing enum.

File: api/proto/info/info.proto (hunk 1 above)
Comment (gkodali-zededa) [reply api/proto/info/info.proto:208]: Will change the name to START_DELAYED

File: api/proto/info/info.proto (hunk 2 above)
Comment (eriknordmark) [reply api/proto/info/info.proto]: Yes. I was confused by the types.go values which are different and mapped to the API.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Pull request merged**